base/compiler/ssair/inlining.jl

```diff
@@ -376,11 +376,28 @@ function ir_inline_item!(compact::IncrementalCompact, idx::Int, argexprs::Vector
             boundscheck = :off
         end
     end
+    if !validate_sparams(sparam_vals)
+        if def.isva
+            nonva_args = argexprs[1:end-1]
+            va_arg = argexprs[end]
+            tuple_call = Expr(:call, TOP_TUPLE, def, nonva_args...)
+            tuple_type = tuple_tfunc(Any[argextype(arg, compact) for arg in nonva_args])
+            tupl = insert_node_here!(compact, NewInstruction(tuple_call, tuple_type, topline))
+            apply_iter_expr = Expr(:call, Core._apply_iterate, iterate, Core._compute_sparams, tupl, va_arg)
+            sparam_vals = insert_node_here!(compact,
+                effect_free(NewInstruction(apply_iter_expr, SimpleVector, topline)))
```

Comment:
this seems like it would add significant cost (and allocations and new DataType's) that were not present before inlining? are we sure this is profitable?

Reply:
Prior to the `fix_va_argexprs!` call that you have above, the argexprs list was required to be a known length and did not have or need a va_arg

Reply:
Addressed in https://github.com/JuliaLang/julia/pull/46700.